cmd/acr/annotate.go

```diff
@@ -0,0 +1,478 @@
+	--artifact-type "application/vnd.microsoft.artifact.lifecycle"
+
+- Annotate all tags that contain the word test in the tag name in the example.azurecr.io registry inside the hello-world 
+repository, after that, remove the dangling manifests in the same repository
+acr annotate -r example --filter "hello-world:\w*test\w*" --annotations "vnd.microsoft.artifact.lifecycle.end-of-life.date=2024-04-09" 
+--artifact-type "application/vnd.microsoft.artifact.lifecycle" --untagged 
+
+- Annotate all tags that contain the word test in the tag name in the example.azurecr.io registry inside the hello-world
+repository. Two annotations need to be applied.
+acr annotate -r example --filter "hello-world:\w*test\w* --annotations "vnd.microsoft.artifact.lifecycle.end-of-life.date=2024-04-09" 
+--annotations "key=value" --artifact-type "application/vnd.microsoft.artifact.lifecycle" 
+
+- Annotate all tags in the example.azurecr.io registry inside the hello-world repository, with 4 annotate tasks running concurrently
+acr annotate -r example --filter "hello-world:.*" --annotations "vnd.microsoft.artifact.lifecycle.end-of-life.date=2024-04-09" 
+--artifact-type "application/vnd.microsoft.artifact.lifecycle" --concurrency 4
+`
+)
+
+var (
+	annotatedConcurrencyDescription = fmt.Sprintf("Number of concurrent annotate tasks. Range: [1 - %d]", maxPoolSize)
+)
+
+// annotateParameters defines the parameters that the annotate command uses (including the registry name, username, and password)
+type annotateParameters struct {
+	*rootParameters
+	filters       []string
+	filterTimeout uint64
+	artifactType  string
+	annotations   []string
+	untagged      bool
+	dryRun        bool
+	concurrency   int
+}
+
+// newAnnotateCmd defines the annotate command
+func newAnnotateCmd(rootParams *rootParameters) *cobra.Command {
+	annotateParams := annotateParameters{rootParameters: rootParams}
+	cmd := &cobra.Command{
+		Use:     "annotate",
+		Short:   "Annotate images in a registry",
+		Long:    newAnnotateCmdLongMessage,
+		Example: annotateExampleMessage,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// This context is used for all the http requests
+			ctx := context.Background()
+			registryName, err := annotateParams.GetRegistryName()
+			if err != nil {
+				return err
+			}
+			loginURL := api.LoginURL(registryName)
+			// An acrClient with authentication is generated, if the authentication cannot be resolved an error is returned.
+			acrClient, err := api.GetAcrCLIClientWithAuth(loginURL, annotateParams.username, annotateParams.password, annotateParams.configs)
+			if err != nil {
+				return err
+			}
+
+			orasClient, err := api.GetORASClientWithAuth(loginURL, annotateParams.username, annotateParams.password, annotateParams.configs)
+			if err != nil {
+				return err
+			} else {
+				fmt.Println("oras auth ok?")
+			}
+			// orasClient.Annotate(context.Background(), "asdf", "asdf", annotateParams.artifactType, map[string]string{})
```

Comment:
can you remove this commented out code?

Reply:
Sorry I went through and cleaned stuff up but looks like I missed a few